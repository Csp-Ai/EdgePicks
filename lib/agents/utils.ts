import fs from 'fs';
import path from 'path';
import { AgentReflection } from '../../types/AgentReflection';

export const pseudoMetric = async (seed: string, mod: number): Promise<number> => {
  const hash = Array.from(seed).reduce((acc, ch) => acc + ch.charCodeAt(0), 0);
  return hash % mod;
};

export function logAgentReflection(agent: string, reflection: AgentReflection) {
  const logFile = path.join(process.cwd(), 'logs', 'agent-reflections.json');
  let entries: any[] = [];
  try {
    entries = JSON.parse(fs.readFileSync(logFile, 'utf8'));
  } catch {
    entries = [];
  }
  entries.push({ agent, ...reflection, timestamp: new Date().toISOString() });
  fs.writeFileSync(logFile, JSON.stringify(entries, null, 2));
}
