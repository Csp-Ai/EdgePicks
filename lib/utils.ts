/**
 * Calculates a contribution for an agent's score based on its weight.
 *
 * @param score - The agent's score.
 * @param weight - The weight applied to the score.
 * @returns The weighted contribution.
 *
 * @example
 * ```ts
 * getContribution(0.8, 0.5);
 * // => 0.4
 * ```
 */
import { type ClassValue, clsx } from "clsx";
import { twMerge } from "tailwind-merge";

export const getContribution = (score: number, weight: number): number => {
  return score * weight;
};

/**
 * Formats an agent name by capitalizing the first character.
 *
 * @param name - The agent name to format.
 * @returns The formatted agent name with the first letter capitalized.
 *
 * @example
 * ```ts
 * formatAgentName('injuryScout');
 * // => 'InjuryScout'
 * ```
 */
export const formatAgentName = (name: string): string =>
  name.charAt(0).toUpperCase() + name.slice(1);

/**
 * Combines CSS class names into a single string, omitting falsey values.
 *
 * @param classes - List of class names or falsey values.
 * @returns A space-separated string of truthy class names.
 *
 * @example
 * ```ts
 * cn('btn', undefined, 'active');
 * // => 'btn active'
 * ```
 */
export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}
