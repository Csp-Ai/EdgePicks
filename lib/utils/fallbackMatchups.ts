import { Matchup } from '../types';

export function getFallbackMatchups(): (Matchup & { useFallback: true })[] {
  return [
    {
      homeTeam: 'Dallas Cowboys',
      awayTeam: 'New York Giants',
      time: '2025-09-07T20:20:00Z',
      league: 'NFL',
      homeLogo: 'https://a.espncdn.com/i/teamlogos/nfl/500/dal.png',
      awayLogo: 'https://a.espncdn.com/i/teamlogos/nfl/500/nyg.png',
      useFallback: true,
      source: 'fallback',
    },
  ];
}
