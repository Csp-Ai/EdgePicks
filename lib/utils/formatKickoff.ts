export function formatKickoff(iso: string, nowMs: number = Date.now()): string {
  const t = Date.parse(iso);
  if (Number.isNaN(t)) return 'TBD';
  const diff = t - nowMs;
  if (diff <= 0) return 'started';
  const min = Math.round(diff / 60000);
  if (min < 60) return `in ${min}m`;
  const hours = Math.floor(min / 60);
  if (hours < 24) return `in ${hours}h`;
  const days = Math.floor(hours / 24);
  return `in ${days}d`;
}

export function formatAbsolute(iso: string, locale?: string): string {
  const d = new Date(iso);
  try {
    return d.toLocaleString(locale ?? undefined, { month: 'short', day: 'numeric', hour: 'numeric', minute: '2-digit' });
  } catch {
    return d.toISOString();
  }
}

