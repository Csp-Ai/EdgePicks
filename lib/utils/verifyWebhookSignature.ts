import { createHmac, timingSafeEqual } from 'crypto';

export function verifyWebhookSignature(payload: string, signature: string, secret: string): boolean {
  const expected = createHmac('sha256', secret).update(payload).digest('hex');
  const sigBuffer = Buffer.from(signature, 'hex');
  const expectedBuffer = Buffer.from(expected, 'hex');
  return sigBuffer.length === expectedBuffer.length && timingSafeEqual(sigBuffer, expectedBuffer);
}
