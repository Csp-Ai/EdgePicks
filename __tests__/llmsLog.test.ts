import fs from 'fs';
import path from 'path';
import crypto from 'crypto';
import { freezeTime } from './utils/freezeTime';

describe('llms log', () => {
  let unfreeze: () => void;
  let restoreRand: () => void;

  beforeAll(() => {
    unfreeze = freezeTime('2024-01-01T00:00:00.000Z');
    const rand = jest.spyOn(Math, 'random').mockReturnValue(0.123456);
    restoreRand = () => rand.mockRestore();
  });

  afterAll(() => {
    unfreeze();
    restoreRand();
  });

  it('writes entry (stable time)', () => {
    const file = path.join(__dirname, '..', 'llms.txt');
    const content = fs.readFileSync(file, 'utf8');
    const hash = crypto.createHash('sha256').update(content).digest('hex');
    expect(hash).toMatchSnapshot();
  });
});
