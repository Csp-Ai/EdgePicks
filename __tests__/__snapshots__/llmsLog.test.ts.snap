// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`llms.txt log matches content hash snapshot 1`] = `"ea1701456bad2501a63bf781387f5e9018528ad123dd1207876f9491ba11e96b"`;
